worker/src/main/java/gov/cms/ab2d/worker/config/RoundRobinBlockingQueue.java

```diff
@@ -0,0 +1,357 @@
+package gov.cms.ab2d.worker.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Predicate;
+
+@Slf4j
+public class RoundRobinBlockingQueue<E> implements BlockingQueue<E> {
+    // The list of categories
+    private List<String> categories = new ArrayList<>();
+    // The individual category queues
+    private Map<String, LinkedList<E>> categoryQueues = new HashMap<>();
+    // The current category index
+    private int currentIndex = 0;
+    // Main lock guarding all access
+    private ReentrantLock lock;
+    // Not empty condition on the lock
+    private final Condition notEmpty;
+    // Total number in the queues
+    private long count = 0;
+
+    RoundRobinBlockingQueue() {
+        log.info("Created Round Robin Blocking Queue");
+        lock = new ReentrantLock();
+        notEmpty = lock.newCondition();
+    }
+
+    // New code to purge items
+    public void removeIfTrue(Predicate<E> op) {
```

Comment:
This method is not a part of the `BlockingQueue` interface or any of its super-interfaces, which means we most likely don't need it. I know it is referenced from `RoundRobinThreadPoolExecutor`, but that class will likely have to go (see other comments).